src/Polly.SharedSpecs/Registry/DefaultPolicyRegistrySpecs.cs

```diff
@@ -0,0 +1,238 @@
+            Policy outPolicy = null;
+
+            _registry.Add(key, policy);
+
+            _registry.Invoking(r => outPolicy = r[key])
+                .ShouldNotThrow<KeyNotFoundException>();
+
+            outPolicy.ShouldBeEquivalentTo(policy);
+        }
+
+        [Fact]
+        public void Should_Not_Throw_While_Retrieving_When_Key_Does_Not_Exist_Using_TryGetValue()
+        {
+            string key = Guid.NewGuid().ToString();
+            Policy outPolicy = null;
+            bool result = false;
+
+            _registry.Invoking(r => result = r.TryGetValue(key, out outPolicy))
+                .ShouldNotThrow();
+
+            result.Should().BeFalse();
+        }
+
+        [Fact]
+        public void Should_Overwrite_Existing_Policy_If_Key_Exists_When_Inserting_Using_Idexer()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            Policy policy_new = Policy.NoOp();
+
+            _registry.Invoking(r => r[key] = policy_new)
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(1);
+
+            Policy output = null;
+            _registry.Invoking(r => output = r[key])
+                .ShouldNotThrow();
+            output.ShouldBeEquivalentTo(policy_new);
+        }
+
+        [Fact]
+        public void Should_Allow_Clearing_Registry()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(1);
+
+            policy = Policy.NoOp();
+            key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(2);
+
+            _registry.Invoking(r => r.Clear())
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(0);
+        }
+
+        [Fact]
+        public void Should_Allow_Removing_Policy()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(1);
+
+            _registry.Invoking(r => r.Remove(key))
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(0);
+        }
+
+        [Fact]
+        public void Should_Check_If_Key_Exists()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            bool result = false;
+            _registry.Invoking(r => result = r.ContainsKey(key))
+                .ShouldNotThrow();
```

Comment:
I have removed `.ShouldNotThrow()` which were redundant unless, as you mentioned, to make a contrast with elsewhere in the test (`Should_not_be_able_to_add_Policy_with_duplicate_key_using_Add`) and where the test explicitly says Should_not_throw (`Should_not_throw_while_retrieving_when_key_does_not_exist_using_TryGetValue`)

PS: I hadn't worked with FluentAssertion much before. Thanks for helping me out with best practices. Really appreciate it.